File: Exam-Portal-UI/src/app/services/question.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './url';

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private http: HttpClient) { }

  public getQuestionByQuiz(qId: any) {
    return this.http.get(`${baseUrl}/question/quiz/?v=${qId}`);
  }

  public addQuestion(question: any) {
    return this.http.post(`${baseUrl}/question/add`, question);
  }

  public getQuestionById(id: any) {
    return this.http.get(`${baseUrl}/question/get?v=${id}`);
  }

  public updateQuestion(question: any) {
    return this.http.put(`${baseUrl}/question/update`, question);
  }

  public deleteQuestion(quesId: any) {
    return this.http.delete(`${baseUrl}/question/delete?v=${quesId}`);
  }
}

File: ExamServer/src/main/java/com/nagarro/service/QuestionService.java
package com.nagarro.service;

import java.util.List;

import com.nagarro.entity.exam.Question;
import com.nagarro.entity.exam.Quiz;

public interface QuestionService {
	Question addQuestion(Question question);
	Question updateQuestion(Question question);
	List<Question> getAllQuestion();
	Question getQuestionById(int id);
	void deleteQuestion(int id);
	List<Question> getQuestionOfQuiz(Quiz quiz);
}
